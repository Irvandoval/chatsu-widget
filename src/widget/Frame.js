import React, { Component } from 'react';

export default class Frame extends Component {

  shouldComponentUpdate() {
    return false;
  }

  render() {
    let dynamicConf = window.chatWidgetOnOpen || {};
    let encodedConf = encodeURIComponent(JSON.stringify({...this.props.conf, ...this.props.dynamicConf}));
    return (
      <iframe
        title='CxChatWidgetiFrame'
        src={`${this.props.iFrameSrc}?id=${this.props.chatWidgetId}&host=${this.props.host}&conf=${encodedConf}`}
        width='100%'
        height={this.props.isMobile ? '94%' : '100%'}
        frameBorder='0'>
      </iframe>
    )
  }
}