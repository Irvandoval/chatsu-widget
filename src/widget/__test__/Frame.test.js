import React from 'react';
import Frame from '../Frame';
import { JSDOM } from 'jsdom';
import { shallow } from 'enzyme';

describe('<Frame />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Frame />);
    expect(tree).toMatchSnapshot();
  });
})