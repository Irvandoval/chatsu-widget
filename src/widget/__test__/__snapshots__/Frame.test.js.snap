// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frame /> matches the snapshot 1`] = `
<iframe
  frameBorder="0"
  height="100%"
  src="undefined?id=undefined&host=undefined&conf=%7B%7D"
  title="CxChatWidgetiFrame"
  width="100%"
/>
`;
